guides/micronaut-data-mongodb/java/src/main/java/example/micronaut/CourseController.java

```diff
@@ -0,0 +1,40 @@
+package example.micronaut;
+
+import example.micronaut.domain.Course;
+import io.micronaut.http.annotation.Controller;
+import io.micronaut.http.annotation.Get;
+import io.micronaut.http.annotation.PathVariable;
+import io.micronaut.http.annotation.Post;
+import io.micronaut.http.annotation.Put;
+
+import java.util.Optional;
+
+@Controller("/course")
```

Comment:
should not the controller be annotated wtih `@ExecuteOn`. I assume the repository actions are blocking.

Reply:
I'll add a service layer